rust/lance/src/dataset/scanner.rs

```diff
@@ -63,6 +66,46 @@ const DEFAULT_BATCH_READAHEAD: usize = 16;
 // Same as pyarrow Dataset::scanner()
 const DEFAULT_FRAGMENT_READAHEAD: usize = 4;
 
+pub struct Ordering {
+    pub ascending: bool,
+    pub nulls_first: bool,
+    pub column: String,
+}
+
+impl Ordering {
+    pub fn asc_nulls_first(column: String) -> Self {
+        Self {
+            ascending: true,
+            nulls_first: true,
+            column,
+        }
+    }
+
+    pub fn asc_nulls_last(column: String) -> Self {
+        Self {
+            ascending: true,
+            nulls_first: false,
+            column,
+        }
+    }
+
+    pub fn desc_nulls_first(column: String) -> Self {
+        Self {
+            ascending: false,
+            nulls_first: true,
+            column,
+        }
+    }
+
+    pub fn desc_nulls_last(column: String) -> Self {
+        Self {
+            ascending: false,
+            nulls_first: false,
+            column,
+        }
+    }
```

Comment:
The nulls first / nulls last behavior here is a little meaningless at the moment since lance files can't really represent null.  I wasn't sure if it was better to:

 * Create a potentially misleading API the mentions nulls even if there is no point
 * Create an API that we know is going to change because it doesn't mention nulls

Reply:
Well we do with string columns, so it's meaningful there for now. Seems good to handle nulls here.